include/particle.h

```diff
@@ -55,6 +55,11 @@ class Particle : public ParticleBase<Tdim> {
   //! \retval status Status of reading HDF5 particle
   bool initialise_particle(const HDF5Particle& particle) override;
 
+  //! Output particle as HDF5 data
+  //! \param[in] particle HDF5 data of particle
+  //! \retval status Status of writing HDF5 particle
+  bool get_particle_data(unsigned phase, HDF5Particle& particle_data) override;
```

Comment:
```suggestion
  HDF5Particle particle_data(unsigned phase) const override;
```

Reply:
Return a HDF5Particle object instead of pass by reference. We don't modify objects passed by reference. To modify we normally use a pointer. In this case returning a value and emplace into a vector is better

Reply:
copy elision would mean the compiler will generate the same assembly code, so its better to say nothing gets modified in the function, we are likely to get a better performance